scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -505,7 +505,18 @@ class FormatOps(
         val afterInfix = style.breakAfterInfix(t)
         if (afterInfix ne Newlines.AfterInfix.keep) {
           if (ft.meta.leftOwner ne app.op) Seq(Split(Space, 0))
-          else InfixSplits(app, ft).getBeforeLhsOrRhs(afterInfix)
+          else {
+            val fullInfixApp = InfixSplits.findEnclosingInfix(app)
+            val fullInfix = fullInfixApp.all
+            val ok = isEnclosedInParens(fullInfix) || fullInfix.parent.exists {
+              case t: Defn.Val => t.rhs eq fullInfix
```

Comment:
```suggestion
              case t: Defn.Val => t.pat eq fullInfix
```
shouldn't we check the pattern and not rhs?